### PR TITLE
Migrations to phpunit

### PR DESCRIPTION
**Pull request by trasher**
Migrations to phpunit (tests that only exists on main).

### CODE REVIEW

**Comment by trasher** [comment]
The `WebhookTest` causes several "risky tests":
```
1) tests\units\WebhookTest::testGetWebhookBody
Test code or tested code closed output buffers other than its own
```

I'm not sure what the problem is exactly (and I do not really want to spend time on that). If that can cause issues with phpunit, I'll revert this one.

**Comment by cconard96** [comment]
> The `WebhookTest` causes several "risky tests":
> 
> ```
> 1) tests\units\WebhookTest::testGetWebhookBody
> Test code or tested code closed output buffers other than its own
> ```
> 
> I'm not sure what the problem is exactly (and I do not really want to spend time on that). If that can cause issues with phpunit, I'll revert this one.

Probably related to the calls to the new API. None of the webhook-specific classes should be doing anything with the output buffer.

**Comment by cconard96** [comment]
If you want to remove the Workflow migration here, I can work on the PHPUnit tests for it and the new API together.

**Comment by trasher** [comment]
> If you want to remove the Webhook migration here, I can work on the PHPUnit tests for it and the new API together.

Migration is done, and it's not a blocker, so we can keep it for now; and get that fixed later. @cedric-anne what do you think?

**Comment by cedric-anne** [comment]
> > If you want to remove the Webhook migration here, I can work on the PHPUnit tests for it and the new API together.
> 
> Migration is done, and it's not a blocker, so we can keep it for now; and get that fixed later. @cedric-anne what do you think?

I think it is not a blocker for this PR, but there is something that has to be fixed anyway, preferably before the GLPI 11.0 release.

**Comment by trasher** [comment]
I created an issue so we do not forget this. Current PR is OK to be reviewed as is then ;)

File: phpunit/functional/ChangeSatisfactionTest.php [hunk 1]
```diff
@@ -36,8 +36,8 @@
 namespace tests\units;
 
 // Force import because of atoum autoloader not working
-require_once 'CommonITILSatisfaction.php';
+require_once __DIR__ . '/../abstracts/CommonITILSatisfaction.php';
```
Comment (cconard96) [inline phpunit/functional/ChangeSatisfactionTest.php:39]: Is this still needed after switching to PHPUnit?

File: phpunit/functional/TicketSatisfactionTest.php [hunk 2]
```diff
@@ -36,8 +36,8 @@
 namespace tests\units;
 
 // Force import because of atoum autoloader not working
-require_once 'CommonITILSatisfaction.php';
+require_once __DIR__ . '/../abstracts/CommonITILSatisfaction.php';
```
Comment (cconard96) [inline phpunit/functional/TicketSatisfactionTest.php:39]: Is this still needed?

File: phpunit/functional/ChangeSatisfactionTest.php (hunk 1 above)
Comment (trasher) [reply phpunit/functional/ChangeSatisfactionTest.php:39]: Yes

File: phpunit/functional/Item_LineTest.php [hunk 3]
```diff
@@ -64,64 +66,63 @@ public function testAdd()
...
+        $this->assertFalse(
+            $item_line->add([
                 'itemtype'  => 'Phone',
                 'items_id'  => $phone_1
             ])
-        )->isIdenticalTo(0);
+        );
 
         $this->hasSessionMessages(ERROR, ['A line is required']);
+        //$this->hasPhpLogRecordThatContains(LogLevel::WARNING, 'A line is required.');
```
Comment (cedric-anne) [inline phpunit/functional/Item_LineTest.php:77]:
```suggestion
```

File: phpunit/functional/Item_LineTest.php [hunk 4]
```diff
@@ -36,10 +36,11 @@
 namespace tests\units;
 
 use DbTestCase;
+use Psr\Log\LogLevel;
```
Comment (cedric-anne) [inline phpunit/functional/Item_LineTest.php:39]:
```suggestion
```

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**